src/main/kotlin/no/nav/familie/ks/sak/common/tidslinje/Tidslinje.kt

```diff
@@ -0,0 +1,123 @@
+package no.nav.familie.ks.sak.common.tidslinje
+
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.mapper
+import java.time.DayOfWeek
+import java.time.LocalDate
+
+enum class TidsEnhet {
+    DAG,
+    UKE,
+    MÅNED,
+    ÅR
+}
+
+/**
+ * En tidslinje består av ulike verdier over tid. Det vil si at en tidslinje kan ha en verdi
+ * A over et tidsintervall og en annen verdi B over et senere tidsintervall. Disse tidsintervallene blir håndert av Periode-klassen.
+ * En tidslinje kan visualiseres som en liste [A, A, A, A, ... , A, B, B, ... , B, C, C, ... , C] og et startstidspunkt,
+ * hvor A, B og C er ulike verdier og hver plass i lista representerer en dag.
+ * Tidslinjer kan innholde uendelige perioder. Det kan ikke være flere perioder med andre verdier etter en uendelig periode.
+ * En tidslinje støtter verdier av typen [Udefinert], [Null] og [PeriodeVerdi]. En verdi er udefinert når vi ikke vet
+ * hva verdien skal være (et hull i tidslinja). En verdi er no.nav.familie.ks.sak.common.tidslinje.Null når vi vet at det ikke finnes noe verdi i dette tidsrommet.
+ */
+open class Tidslinje<T>(
+    var startsTidspunkt: LocalDate,
+    perioder: List<TidslinjePeriode<T>>,
+    var tidsEnhet: TidsEnhet = TidsEnhet.DAG
+) {
+
+    var innhold: List<TidslinjePeriode<T>> = emptyList()
+        set(verdi) {
+            field = this.lagInnholdBasertPåPeriodelengder(verdi)
+        }
+    val foreldre: MutableList<Tidslinje<Any>> = mutableListOf()
+    var tittel = ""
+
+    init {
+        this.innhold = perioder
+        startsTidspunkt = when (tidsEnhet) {
+            TidsEnhet.ÅR -> startsTidspunkt.withDayOfYear(1)
+            TidsEnhet.MÅNED -> startsTidspunkt.withDayOfMonth(1)
+            TidsEnhet.UKE -> startsTidspunkt.with(DayOfWeek.MONDAY)
+            TidsEnhet.DAG -> startsTidspunkt
+        }
+    }
+
+    /**
+     * Kalkulerer slutttidspunkt som en LocalDate.
+     * Funkjsonen returnerer den siste dagen som er med i tidslinja
+     * Om tidslinja er uendelig, kastes det et unntak
+     */
+    fun kalkulerSluttTidspunkt(): LocalDate {
+        val antallTidsEnheter: Int = this.innhold.sumOf { it.lengde }
+        val sluttTidspunkt = this.startsTidspunkt.plus(antallTidsEnheter.toLong() - 1, mapper[this.tidsEnhet])
+
+        return when (this.tidsEnhet) {
+            TidsEnhet.ÅR -> sluttTidspunkt.withDayOfYear(sluttTidspunkt.lengthOfYear())
```

Comment:
```suggestion
            TidsEnhet.ÅR -> sluttTidspunkt.with(TemporalAdjusters.lastDayOfYear())```

Reply:
Kanskje den er enklere å forstå, samme gjelder for lastDayOfMonth

Reply:
Fikset i [Bruk TemporalAdjusters for å finne siste dag i måned og år](https://github.com/navikt/familie-ks-sak/pull/80/commits/0a9c68aefd4855adff1e7fad5aa39d6771dc0160)